src/vtp/script_libs/cast_ballot_lib.py

```diff
@@ -0,0 +1,334 @@
+                        f"you supplied {len(selections)}"
+                    )
+                    continue
+                #            import pdb; pdb.set_trace()
+                if num in validated_selections:
+                    errors.append(
+                        f"The selection {num} was supplied more than once.  "
+                        "Each selection can only be supplied once."
+                    )
+                    continue
+                validated_selections.append(num)
+            if errors:
+                err_string = "\n".join(errors)
+                raise ValueError(
+                    "Warning - you selections have the following errors\n"
+                    f"{err_string}"
+                )
+            # if still here, set the selection
+            try:
+                # Since it is possible to self adjudicate a contest, always
+                # explicitly clear the selection before adding
+                the_ballot.clear_selection(the_contest)
+                for sel in validated_selections:
+                    the_ballot.add_selection(the_contest, sel)
+            # pylint: disable=broad-except
+            except Exception:
+                # blow out of the internal pyinputplus try/catch
+                traceback.print_exc()
+                sys.exit(1)
+
+        if tally == "plurality":
+            if max_votes > 1:
+                prompt = (
+                    f"Please enter the numbers for your choices (max={max_votes}) "
+                    "separated by spaces: "
+                )
+            else:
+                prompt = "Please enter the number for your choice: "
+            pyinputplus.inputCustom(validate_multichoice, prompt=prompt, blank=True)
+        else:
+            # Then prompt for input
+            prompt = "Please enter in rank order the numbers of your choices separated by spaces: "
+            pyinputplus.inputCustom(validate_multichoice, prompt=prompt, blank=True)
+
+    def loop_over_contests(self, a_ballot):
+        """Will loop over the contests in a ballot and either ask the user
+        for a choice or if in demo mode will randomly choose one.
+        """
+        contests = Contests(a_ballot)
+        total_contests = contests.len()
+        count = 0
+        contest_uids = []
+        for contest in contests:
+            contest_uids.append(contest.get("uid"))
+            if self.parsed_args.demo_mode:
+                self.make_random_selection(a_ballot, contest)
+            else:
+                # Display the tally type and choices and allow the user to manually
+                # enter something.  Might as well validate legal selections (in
+                # this demo) as that is the long-term VTP vision.
+                count += 1
+                self.get_user_selection(a_ballot, contest, count, total_contests)
+        if not self.parsed_args.demo_mode:
```

Comment:
Similarly this block won't work outside of the CLI.

Reply:
Ditto regarding the preceding conversation